ray-curator/api-design.md

```diff
@@ -0,0 +1,198 @@
+# NeMo Curator Ray API Design
+
+**Status:** Pre Release
+
+## Table of Contents
+
+1. [Background](#background)
+2. [Design Principles](#design-principles)
+3. [Core Components](#core-components)
+   - [Tasks](#tasks)
+   - [Stages](#stages)
+   - [Pipelines](#pipelines)
+   - [Executors](#executors)
+4. [Examples](#examples)
+
+## Background
+
+### Current State
```

Comment:
Goal is to remove this section when we merge `ray-api` to main, right?

Reply:
I kept this for developers and SAs who are moving from curator to ray now